src/shipit_code_coverage/shipit_code_coverage/chunk_mapping.py

```diff
@@ -14,52 +15,171 @@
 logger = get_logger(__name__)
 
 
+PLATFORMS = ['linux', 'windows']
+# TODO: Calculate this dinamically when https://github.com/klahnakoski/ActiveData-ETL/issues/40 is fixed.
+TEST_COVERAGE_SUITES = ['reftest', 'web-platform', 'mochitest', 'xpcshell', 'jsreftest', 'crashtest']
+
+
+def get_suites(revision):
+    r = requests.post('https://activedata.allizom.org/query', json={
+        'from': 'unittest',
+        'where': {'and': [
+            {'eq': {'repo.branch.name': 'mozilla-central'}},
+            {'eq': {'repo.changeset.id12': revision[:12]}},
+            {'or': [
+                {'prefix': {'run.key': 'test-linux64-ccov'}},
+                {'prefix': {'run.key': 'test-windows10-64-ccov'}}
+            ]}
+        ]},
+        'limit': 500000,
+        'groupby': ['run.suite']
+    })
+
+    suites_data = r.json()['data']
+
+    return [e[0] for e in suites_data]
+
+
+# Retrieve chunk -> tests mapping from ActiveData.
+def get_tests_chunks(revision, platform, suite):
+    if platform == 'linux':
+        run_key_prefix = 'test-linux64-ccov'
+    elif platform == 'windows':
+        run_key_prefix = 'test-windows10-64-ccov'
+
+    r = requests.post('https://activedata.allizom.org/query', json={
+        'from': 'unittest',
+        'where': {'and': [
+            {'eq': {'repo.branch.name': 'mozilla-central'}},
+            {'eq': {'repo.changeset.id12': revision[:12]}},
+            {'eq': {'run.suite': suite}},
+            {'prefix': {'run.key': run_key_prefix}},
+        ]},
+        'limit': 50000,
+        'select': ['result.test', 'run.key']
+    })
+
+    return r.json()['data']
+
+
+def group_by_20k(data):
+    groups = defaultdict(list)
+    total_count = 0
+
+    for elem, count in data:
+        total_count += count
+        groups[total_count // 20000].append(elem)
+
+    return groups.values()
+
+
+def get_test_coverage_suites():
+    r = requests.post('https://activedata.allizom.org/query', json={
+        'from': 'coverage',
+        'where': {'and': [
+            {'gte': {'repo.push.date': {'date': 'today-week'}}},
```

Comment:
I need to add mozilla-central as a branch here, otherwise we'll gather try results too.